Accompany/Controllers/Core/Home/AddNoteDetailsTableViewController.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  SemesterDetailsTableViewController.swift
+//  Accompany
+//
+//  Created by Kelbin David on 2022-02-10.
+//
+
+import UIKit
+import SnapKit
+
+protocol AddNoteDetailsTableViewControllerDelegate {
+  func add(todo: ToDo)
+  func edit(todo: ToDo)
+}
+
+class AddNoteDetailsTableViewController: UITableViewController {
+  
+  var saveBarButton: UIBarButtonItem = {
+    let barButton = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(saveButtonTapped))
+    return barButton
+  }()
+  
+  var todo: ToDo?
+  var delegate: AddNoteDetailsTableViewControllerDelegate?
+  var isCompleteButton: UIButton?
+  var toDoTitleTextField: UITextField?
+  var noteTextView: UITextView?
+  let buttonIndexPath = IndexPath(row: 0, section: 0)
+  let noteIndexPath = IndexPath(row: 0, section: 1)
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    self.navigationItem.rightBarButtonItem = saveBarButton
+    if todo != nil {
+      navigationItem.title = "Edit To Do"
+    } else {
+      navigationItem.title = "Add To Do"
+    }
+    tableView.register(AddNoteCellOneTableViewCell.self, forCellReuseIdentifier: "cell")
+    tableView.register(AddNoteCellTwoTableViewCell.self, forCellReuseIdentifier: "cell2")
+    tableView.isUserInteractionEnabled = true
+    let notificationCenter = NotificationCenter.default
+    notificationCenter.addObserver(self, selector: #selector(textViewDidChange), name: UITextView.textDidChangeNotification, object: nil)
+    updateSaveButton()
+  }
+  
+  @objc func saveButtonTapped(){
+    let isComplete = isCompleteButton!.isSelected
+    let toDoTitle = toDoTitleTextField!.text!
+    let note = noteTextView?.text ?? ""
+    let newToDo = ToDo(title: toDoTitle, isCompleted: isComplete, note: note)
+    if todo != nil {
+      delegate?.edit(todo: newToDo)
+    } else {
+      delegate?.add(todo: newToDo)
+    }
+    self.navigationController?.popViewController(animated: true)
+  }
+
+  func updateSaveButton(){
```

Comment:
Please refactor the name of the function to updateSaveButtonState.
```suggestion
  func updateSaveButtonState() {
```